dashboard/agent.py

```diff
@@ -92,10 +86,83 @@ async def _check_parent():
                     dashboard_consts.
                     DASHBOARD_AGENT_CHECK_PARENT_INTERVAL_SECONDS)
 
+        check_parent_task = asyncio.create_task(_check_parent())
+
+        # Create an aioredis client for all modules.
+        try:
+            self.aioredis_client = await dashboard_utils.get_aioredis_client(
+                self.redis_address, self.redis_password,
+                dashboard_consts.CONNECT_REDIS_INTERNAL_SECONDS,
+                dashboard_consts.RETRY_REDIS_CONNECTION_TIMES)
+        except (socket.gaierror, ConnectionRefusedError):
+            logger.error(
+                "Dashboard agent suicide, "
+                "Failed to connect to redis at %s", self.redis_address)
+            sys.exit(-1)
+
+        # Create a http session for all modules.
+        self.http_session = aiohttp.ClientSession(
+            loop=asyncio.get_event_loop())
+
+        # Start a grpc asyncio server.
+        await self.server.start()
+
         modules = self._load_modules()
-        await asyncio.gather(_check_parent(),
+
+        # Http server should be initialized after all modules loaded.
+        app = aiohttp.web.Application()
+        app.add_routes(routes=routes.bound_routes())
+
+        # Enable CORS on all routes.
+        cors = aiohttp_cors.setup(
```

Comment:
Previously, we disabled cors when it was dev mode because frontend usually wants to start a react server. We aren't doing that anymore?

Reply:
Currently, the `CORS` is for the new `log` module. Each agent starts a http server for log file access. Allow new dashboard frontend to access log file as static resource directly.

Reply:
`dashboard` has the same access level with `agent`. We can access all routes of `dashboard`, why disallow access to all routes of `agent`?

The `log`http server on each `agent` can reduce the load of the `dashboard`.